core/sql/generator/GenRelScan.cpp

```diff
@@ -216,41 +217,56 @@ int HbaseAccess::createAsciiColAndCastExpr(Generator * generator,
     }
 
   if (newGivenType->getTypeQualifier() == NA_CHARACTER_TYPE &&
-      (CmpCommon::getDefaultString(HIVE_FILE_CHARSET) == "GBK" || CmpCommon::getDefaultString(HIVE_FILE_CHARSET) == "gbk") && CmpCommon::getDefaultString(HIVE_DEFAULT_CHARSET) == "UTF8" )
-        needTranslate = TRUE;
-
+      (CmpCommon::getDefaultString(HIVE_FILE_CHARSET) == "GBK" || 
```

Comment:
Don't we case-shift these things before we get this far? Otherwise we have to worry about variants such as Gbk, gBK, etc.